addons/mail/static/src/components/chatter/chatter_tests.js

```diff
@@ -453,6 +462,346 @@ QUnit.test('do not post message with "Enter" keyboard shortcut', async function
+        id: 10,
+        email_cc: "john@test.be",
+        partner_ids: [100],
+    });
+    await this.start ();
+    const chatter = this.env.models['mail.chatter'].create({
+        threadId: 10,
+        threadModel: 'res.fake',
+    });
+    await this.createChatterComponent({ chatter });
+    await afterNextRender(() =>
+        document.querySelector(`.o_ChatterTopbar_buttonSendMessage`).click()
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_ComposerSuggestedRecipientList',
+        "Should display a list of suggested recipients after opening the composer from 'Send message' button"
+    );
+});
+
+QUnit.test("suggested recipient list should not contain a \"show more\" button when the list is <= 3 suggestions", async function (assert) {
+    assert.expect(1);
+
+    this.data['res.partner'].records.push({
+        display_name: "John Jane",
+        email: "john@jane.be",
+        id: 100,
+    });
+    this.data['res.fake'].records.push({
+        id: 10,
+        email_cc: "john@test.be",
+        partner_ids: [100],
+    });
+    await this.start ();
+    const chatter = this.env.models['mail.chatter'].create({
+        threadId: 10,
+        threadModel: 'res.fake',
+    });
+    await this.createChatterComponent({ chatter });
+    await afterNextRender(() =>
+        document.querySelector(`.o_ChatterTopbar_buttonSendMessage`).click()
+    );
+    assert.containsNone(
+        document.body,
+        '.o_ComposerSuggestedRecipientList_showMore',
+        "suggested recipient list should not contain a \"show more\" button when the list is <= 3 suggestions"
+    );
+});
+
+QUnit.test("display reason for suggested recipient on mouse over", async function (assert) {
+    assert.expect(2);
+
+    this.data['res.partner'].records.push({
+        display_name: "John Jane",
+        email: "john@jane.be",
+        id: 100,
+    });
+    this.data['res.fake'].records.push({
+        id: 10,
+        email_cc: "john@test.be",
+        partner_ids: [100],
+    });
+    await this.start();
+    const chatter = this.env.models['mail.chatter'].create({
+        threadId: 10,
+        threadModel: 'res.fake',
+    });
+    await this.createChatterComponent({ chatter });
+    await afterNextRender(() =>
+        document.querySelector(`.o_ChatterTopbar_buttonSendMessage`).click()
+    );
+    const title = document.querySelector('.o_ComposerSuggestedRecipient:not([data-partner-id])').getAttribute('title');
+    assert.strictEqual(
+        title,
+        "Add as recipient and follower (reason: CC email)",
+        "display reason for suggested recipient without partner on mouse over",
+    );
+    const partnerTitle = document.querySelector('.o_ComposerSuggestedRecipient[data-partner-id="100"]').getAttribute('title');
+    assert.strictEqual(
+        partnerTitle,
+        "Add as recipient and follower (reason: Email partner)",
+        "display reason for suggested recipient with partner on mouse over",
+    );
+});
```

Comment:
Should be 2 different tests. One for email partner reason and another for cc email reason.

Reply:
Sound a bit silly now since both test will be testing the same code. The partner doesn't play any role in the reason.
So maybe you are right an we only need one test after all.

Reply:
Test are there to verify that code does indeed something. In many ways, the code itself should be perceived as a black box. So the argument that _"it will be testing the same code"_ is irrelevant for deciding to group things that are (even slightly) different.

I'm good with just 1 test for just partner though. The "without partner" case is not tested, therefore bugs may happen and we didn't catch them. Thing is the more test we add, the more maintenance it adds. It's also unreasonable to have no tests, or full coverage with tests (code or functional coverage: full functional coverage with current code would imply hundreds of thousands of unit tests..).

The test "without partner" could maybe be introduced later in case we have specific bugs on them. IMO the main purpose of test is to built confidence to a satisfying level. In other words, if we are confident enough with just the code, I'm totally good not having any tests.